// The case below contains a typo (test.value instead of test.item)
// This caused a compiler crash, and should not.
// EXIT=2
generic [type item] type test is record with
  a : item
  b : item

function test (n : test.value) return test is
   result.a := n
   result.b := test.value(0)

K : test(3)
L : test(3.5)

function tart (n : test.item) return test is
   result.a := test.item(3)
   result.b := test.value(4)

M : tart(3)
